### PR TITLE
docs: add entry for noShorthandPropertyOverrides to CHANGELOG.md

### PR DESCRIPTION
**Pull request by neokidev**
## Summary

- Add changelog entry for the new feature introduced in #2958
- Fix typo in link to noUnknownUnit documentation in `CHANGELOG.md`

## Test Plan

- The corrected link now points to the correct documentation for Biome

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -465,7 +469,7 @@ New rules are incubated in the nursery group. Once stable, we promote them to a
 - Add [nursery/noUnknownMediaFeatureName](https://biomejs.dev/linter/rules/no-unknown-media-feature-name). [#2751](https://github.com/biomejs/biome/issues/2751) Contributed by @Kazuhiro-Mimaki
 - Add [nursery/noUnknownProperty](https://biomejs.dev/linter/rules/no-unknown-property). [#2755](https://github.com/biomejs/biome/pull/2755) Contributed by @chansuke
 - Add [nursery/noUnknownSelectorPseudoElement](https://biomejs.dev/linter/rules/no-unknown-selector-pseudo-element). [#2655](https://github.com/biomejs/biome/issues/2655) Contributed by @keita-hino
-- Add [nursery/noUnknownUnit](https://biomejs.dev/linter/rules/no-unknwon-unit). [#2535](https://github.com/biomejs/biome/issues/2535) Contributed by @neokidev
+- Add [nursery/noUnknownUnit](https://biomejs.dev/linter/rules/no-unknown-unit). [#2535](https://github.com/biomejs/biome/issues/2535) Contributed by @neokidev
```
Comment (neokidev) [inline CHANGELOG.md:472]: I noticed and fixed the typo by chance. Should I split this into a separate PR?

**Review by ematipico — APPROVED** [review]
